packages/react-hooks/src/ConnectionProvider.tsx

```diff
@@ -30,12 +30,22 @@ export const useConnectionById = (id: string): ConnectionRecord | undefined => {
   return connections.find((c: ConnectionRecord) => c.id === id)
 }
 
-export const useConnectionByState = (state: DidExchangeState): ConnectionRecord[] => {
+export const useConnectionByState = (
+  state: DidExchangeState | DidExchangeState[],
+  invertSearch = false
+): ConnectionRecord[] => {
+  let states = typeof state === 'string' ? [state] : state
+
+  if (invertSearch) {
+    states = Object.values(DidExchangeState).filter((v) => !states.includes(v))
+  }
   const { records: connections } = useConnections()
-  const filteredConnections = useMemo(
-    () => connections.filter((c: ConnectionRecord) => c.state === state),
-    [connections, state]
-  )
+
+  const filteredConnections = states
+    .map((filterState: DidExchangeState) =>
+      useMemo(() => connections.filter((c: ConnectionRecord) => c.state === filterState), [connections, filterState])
+    )
+    .flat()
```

Comment:
Also this loops through the connections list for each of the passed states. What if we filter it like this: 

```
const filteredConnections = useMemo(() => connections.filter(c => states.includes(c.state)))
```

This can save quite some loops if we're filtering for multiple states (especially with the inverse filter)

Reply:
Don't forget here to add the dependency array :).